README.md
# masscan-web-ui
MASSCAN Web UI

Here's a quick guide to get started:

* Install and setup your web server, copy over the MASSCAN web-ui files:

```
root@kali:~# apt-get install apache2 php libapache2-mod-php php-mysqli php-xml mysql-server
root@kali:~# git clone https://github.com/dustinwloring1988/masscan-web-ui
root@kali:~# mv masscan-web-ui/* /var/www/html/
root@kali:~# cd /var/www/html/
```

* Create a database, user, then import database.

```
root@kali:/var/www/html# mysql -u root -p
Enter password: 
Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> create database masscan;
Query OK, 1 row affected (0.01 sec)

mysql> CREATE USER 'masscan'@'localhost' IDENTIFIED BY 'changem3';
Query OK, 0 rows affected (0.00 sec)

mysql> GRANT ALL PRIVILEGES ON masscan.* TO 'masscan'@'localhost';
Query OK, 0 rows affected (0.01 sec)

mysql> exit
Bye
root@kali:/var/www/html# mysql -u root -p masscan < db-structure.sql 
Enter password: 
root@kali:/var/www/html# rm db-structure.sql index.html README.md
```

Update the web config file with the database information:

```
nano config.php
define('DB_DRIVER',	    'mysql');
define('DB_HOST',       'localhost');
define('DB_USERNAME',   'masscan');
define('DB_PASSWORD',   'changem3');
define('DB_DATABASE',   'masscan');
```

Now you can use masscan to scan your targets, while specifying an XML output. More information about banner grabbing with masscan can be found at https://github.com/robertdavidgraham/masscan#banner-checking.

```
masscan 10.0.0.0/8 -p80,21,53 --banners --source-ip 10.0.0.2 --max-rate 1000000 -oX scan-01.xml
```

Now let's import some scan results. In this example we imported the results of two class A scans:

```
root@kali:/var/www/html# ls -l scan*
-rw-r--r-- 1 root root 212929324 Dec  1 13:23 scan-01.xml
-rw-r--r-- 1 root root 700816226 Dec  1 13:55 scan-02.xml
root@kali:/var/www/html# php import.php scan-01.xml 

Do you want to clear the database before importing (yes/no)?: yes

Clearing the db
Reading file
Parsing file
Processing data (This may take some time depending on file size)

Summary:
Total records:738279
Inserted records:738279
Took about:3 minutes,18 seconds
root@kali:/var/www/html# php import.php scan-02.xml 

Do you want to clear the database before importing (yes/no)?: no
Reading file
Parsing file
Processing data (This may take some time depending on file size)

Summary:
Total records:2411974
Inserted records:2411974
Took about:9 minutes,41 seconds
root@kali:/var/www/html# 
```
A total of more than 3 million results are now easily searchable. 


## Scripts included:
* Bootstrap
* Font Awesome
* jQuery-Autocomplete
* FullCalendar
* Charts.js
* Bootstrap Colorpicker
* Cropper
* dataTables
* Date Range Picker for Bootstrap
* Dropzone
* easyPieChart
* ECharts
* bootstrap-wysiwyg
* Flot - Javascript plotting library for jQuery.
* gauge.js
* iCheck
* jquery.inputmask plugin
* Ion.RangeSlider
* jQuery
* jVectorMap
* moment.js
* Morris.js - pretty time-series line graphs
* PNotify - Awesome JavaScript notifications
* NProgress
* Pace
* Parsley
* bootstrap-progressbar
* select2
* Sidebar Transitions - simple off-canvas navigations
* Skycons - canvas based wather icons
* jQuery Sparklines plugin
* switchery - Turns HTML checkbox inputs into beautiful iOS style switches
* jQuery Tags Input Plugin
* Autosize - resizes text area to fit text
* validator - HTML from validator using jQuery
* jQuery Smart Wizard

## Other templates and useful resources
* [Free Bootstrap Admin Templates](https://colorlib.com/wp/free-bootstrap-admin-dashboard-templates/ "Bootstrap Admin Templates on Colorlib") - List of the best Free Bootstrap admin dashboard templates that are available for free for personal and commercial use.
* [Free Admin Templates](https://colorlib.com/wp/free-html5-admin-dashboard-templates/ "List of free HTML based admin templates by Colorlib") - Long list of the best free HTML5 powered admin dashboard templates. Available for personal and commercial use.
* [Angular Templates](https://colorlib.com/wp/angularjs-admin-templates/ "Angular Admin Templates on Colorlib") - List of the most popular admin templates based on AngularJS.
* [HTML Admin Templates](https://colorlib.com/wp/html-admin-templates/ "Material Design Admin Templates on Colorlib") - Most of these templates are based on AngularJS and uses a stunning Material design.
* [Bootstrap Admin Templates](https://colorlib.com/wp/bootstrap-admin-templates/ "List of Premium Bootstrap Admin Templates by Colorlib") - List of premium Bootstrap admin templates that uses a minimal flat or material design. Majority of these themes uses AngularJS but HTML5 versions are also available.
* [WordPress Admin Templates](https://colorlib.com/wp/wordpress-admin-dashboard-themes-plugins/ "List of WordPress Admin Dashboard Templates and Plugins by Colorlib") - List of the best WordPress admin dashboard templates and plugins that will add a personal touch to your WordPress dashboard.
* [WordPress Themes](https://colorlib.com/wp/free-wordpress-themes/ "List of Free WordPress themes by Colorlib") - A huge selection of the best free WordPress themes that are all licensed under GPL and are available for personal and commercial use without restrictions.

## License information
Gentelella is licensed under The MIT License (MIT). Which means that you can use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software. But you always need to state that Colorlib is the original author of this template.

Project is developed and maintained by [Colorlib](https://colorlib.com/ "Colorlib - Make Your First Blog") and Aigars Silkalns


changelog.md
# NMapWorld Changelog

### Note

Earlier there were no changelog at all and we have introduced one now and we will start from version 1.0.0. However, keep in mind that this is far from being first version as there have been dozens of commits.

### 1.0.0 - 26.01.2021

* Fixed dataTables
* Added new dataTable variations

This version is tested but we would recommend not to use it in production right away. Please install it for testing purposes and report back if there are any problems to be fixed.


includes/setup.php
<?php include DOC_ROOT.'/header.php';?>
    <div class="container errorPage">
        <div class="row">
            <div class="col-md-12">
                <div class="jumbotron">
                    <h2>Setup MASSCAN WEB UI</h2>
                    <?php
                        if (!extension_loaded('simplexml')) {
                            ?>
                                <div class="alert alert-danger">
                                    <p><i class="glyphicon glyphicon-info-sign"></i> PHP-XML package is missing!</p>
                                    <p>To install PHP_XML package on Kali linux type:</p>
                                    <pre class="shell">
root@kali:~# sudo apt-get install php-xml
                                    </pre>
                                    <p>Don't forget to restart webserver after installing.</p>
                                </div>
                            <?php
                        }
                        if (!extension_loaded('pdo') ) {
                            ?>
                            <div class="alert alert-danger">
                                <p><i class="glyphicon glyphicon-info-sign"></i> PHP-PDO package is missing!</p>
                                <p>To install PHP_PDO package on Kali linux type:</p>
                                <pre class="shell">
root@kali:~# sudo apt-get install php-pdo
                                    </pre>
                                <p>Don't forget to restart webserver after installing.</p>
                            </div>
                            <?php
                        }
                        ?>
                        <div class="alert alert-danger">
                            <h4><i class="glyphicon glyphicon-exclamation-sign"></i> Error message</h4>
                            <p><?php echo $pdoe->getMessage ();?>.</p>
                        </div>
                    <p>It's easy and fast. Just follow instructions and setup will be done in a few minutes.</p>
                    <ul class="nav nav-tabs" role="tablist">
                        <li role="presentation" class="<?php if (DB_DRIVER !== 'pgsql') echo 'active'; ?>"><a href="#mysql" aria-controls="home" role="tab" data-toggle="tab">MySQL</a></li>
                        <li role="presentation" class="<?php if (DB_DRIVER == 'pgsql') echo 'active'; ?>"><a href="#pgsql" aria-controls="profile" role="tab" data-toggle="tab">PostgreSQL</a></li>
                    </ul>

                    <div class="tab-content">
                        <div role="tabpanel" class="tab-pane<?php if (DB_DRIVER !== 'pgsql') echo ' active'; ?>" id="mysql">
                            <br>
                            <p>Make sure that you have installed php-mysqli package. If not, open terminal and type:</p>
                            <pre class="shell">
root@kali:~# apt-get install php-mysqli
</pre>
                            <p>Than go to the document root of application by typing:</p>
                            <pre class="shell">
root@kali:~# cd <?php echo DOC_ROOT;?>
</pre>
                            <p>Edit config.php and provide MySql host, user, password and database information. Type:</p>
                            <pre class="shell">
root@kali:<?php echo DOC_ROOT;?># nano config.php
</pre>
                            <p>and change the default values if necessary:</p>
                            <pre>
define('DB_DRIVER',     'mysql');
define('DB_HOST',       '<?php echo DB_HOST;?>');
define('DB_USERNAME',	'<?php echo DB_USERNAME;?>');
define('DB_PASSWORD', 	'<?php echo DB_PASSWORD;?>');
define('DB_DATABASE', 	'<?php echo DB_DATABASE;?>');
</pre>
                            <p class="alert alert-info">If you change default database params in config.php before you continue with setup, in order for the changes to take effect refresh this page (press F5 or click <a href="./">here</a>) and you will have updated all commands later in this page.</p>
                            <p>Next step is to login to MySql and create the database:</p>
                            <pre class="shell">
root@kali:<?php echo DOC_ROOT;?># mysql -u root -p
Enter password:
Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.
mysql>
</pre>
                            <p>Create the database by executing following command:</p>
                            <pre class="shell">
mysql> create database <?php echo DB_DATABASE;?>;
Query OK, 1 row affected (0.01 sec)
</pre>
                            <p>Now create new user:</p>
                            <pre class="shell">
mysql> CREATE USER '<?php echo DB_USERNAME;?>'@'<?php echo DB_HOST;?>' IDENTIFIED BY '<?php echo DB_PASSWORD;?>';
Query OK, 0 rows affected (0.00 sec)
</pre>
                            <p>Give the newly created user <strong><?php echo DB_USERNAME;?></strong> privileges for database <strong><?php echo DB_DATABASE;?></strong>:</p>
                            <pre class="shell">
mysql> GRANT ALL PRIVILEGES ON <?php echo DB_DATABASE;?>.* TO '<?php echo DB_USERNAME;?>'@'<?php echo DB_HOST;?>';
Query OK, 0 rows affected (0.01 sec)
</pre>
                            <p>Finally quit MySql by typing:</p>
                            <pre class="shell">
mysql> exit
Bye
</pre>
                            <p>and you are done.</p>
                            <p>When you finished with all the above, you can continue with installation by pressing F5 or clicking <a href="./">here</a>.</p>
                        </div>
                        <div role="tabpanel" class="tab-pane<?php if (DB_DRIVER == 'pgsql') echo ' active'; ?>" id="pgsql">
                            <br>
                            <p>Make sure that you have installed php-pgsql package. If not, open terminal and type:</p>
                            <pre class="shell">
root@kali:~# apt-get install php-pgsql
</pre>
                            <p>Than go to application document root:</p>
                            <pre class="shell">
root@kali:~# cd <?php echo DOC_ROOT;?>
</pre>
                            <p>Edit config.php and provide PostgreSQL host, user, password and database information. Type:</p>
                            <pre class="shell">
root@kali:<?php echo DOC_ROOT;?># nano config.php
</pre>
                            <p>and change the default values if necessary:</p>
                            <pre>
define('DB_DRIVER',     'pgsql');
define('DB_HOST',       '<?php echo DB_HOST;?>');
define('DB_USERNAME',	'<?php echo DB_USERNAME;?>');
define('DB_PASSWORD', 	'<?php echo DB_PASSWORD;?>');
define('DB_DATABASE', 	'<?php echo DB_DATABASE;?>');
</pre>
                            <p class="alert alert-info">If you change default database params in config.php before you continue with setup, in order for the changes to take effect refresh this page (press F5 or click <a href="./">here</a>) and you will have updated all commands later in this page.</p>
                            <p>Next step is to login to PostgreSQL and create user and database :</p>
                            <pre class="shell">
root@kali:<?php echo DOC_ROOT;?># su postgres
postgres@kali:/home$
</pre>
                            <p>Than log into PostgreSQL server:</p>
                            <pre class="shell">
root@kali:/home$ psql
psql (9.5.7)
Type "help" for help.
postgres=#
</pre>
                            <p>Now, create new user:</p>
<pre class="shell">
postgres=# create user masscan;
CREATE ROLE
postgres=#
</pre>
<p>Generate password for newly create user:</p>
                            <pre class="shell">
postgres=# alter user masscan with password 'changem3';
ALTER ROLE
</pre>
                            <p>Now create database:</p>
                            <pre class="shell">
postgres=# create database masscan;
CREATE DATABASE
</pre>
                            <p>and finally add permissions:.</p>
                            <pre class="shell">
postgres=# grant all privileges on database masscan to masscan;
GRANT
</pre>
                            <p>Finally, exit PostgreSQL:</p>
                            <pre class="shell">
postgres=# \q
postgres@kali:/home$ exit
root@kali:/home$
</pre>
                            <p>and refresh this page.</p>
                        </div>
                    </div>
                </div> <!-- end of .jumbotron -->
            </div> <!-- end .col-md-12 -->
        </div> <!-- end of .row -->
    </div> <!-- end of .container -->
<?php include DOC_ROOT.'footer.php';
